Source/CarthageKit/Project.swift

```diff
@@ -609,6 +605,54 @@ private func filesInDirectory(directoryURL: NSURL, typeIdentifier: String) -> Si
 		}
 }
 
+/// Sends the URL for the Info.plist of the specified Framework.
+private func infoPlistForFramework(frameworkURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
+	return filesInDirectory(frameworkURL, "com.apple.property-list")
+		|> filter { plist in plist.lastPathComponent == "Info.plist" }
+		|> take(1)
+}
+
+/// Sends the platform specified in the given Info.plist.
+private func platformForInfoPlist(plistURL: NSURL) -> SignalProducer<Platform, CarthageError> {
+	return SignalProducer(value: plistURL)
+		|> startOn(QueueScheduler(name: "org.carthage.CarthageKit.Project.platformForInfoPlist"))
+		|> tryMap { URL in
+			var error: NSError?
+			if let data = NSData(contentsOfURL: URL, options: nil, error: &error) {
+				return .success(data)
+			}
+			return .failure(CarthageError.ReadFailed(URL, error))
+		}
```

Comment:
This might be simplified as follows:

```swift
SignalProducer.try { () -> Result<NSData, CarthageError> in
    var error: NSError?
    if let data = NSData(contentsOfURL: plistURL, options: nil, error: &error) {
        return .success(data)
    }
    return .failure(.ReadFailed(plistURL, error))
}
|> startOn(QueueScheduler(name: "org.carthage.CarthageKit.Project.platformForInfoPlist"))
```